docs/data/data-grid/server-side-data/lazy-loading.md

```diff
@@ -2,14 +2,111 @@
 title: React Server-side lazy loading
 ---
 
-# Data Grid - Server-side lazy loading [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')🚧
+# Data Grid - Server-side lazy loading [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
 
 <p class="description">Row lazy-loading with server-side data source.</p>
 
+Lazy Loading changes the way pagination works by removing page controls and loading data dynamically (in a single list) as the user scrolls through the grid.
+
+It is enabled by adding `unstable_lazyLoading` prop in combination with `unstable_dataSource` prop.
+
+Initially, the first page data is fetched and displayed in the grid. What triggers the loading of next page data depends on the value of the total row count.
+
+If the total row count is known, the Data Grid gets filled with skeleton rows and fetches more data if one of the skeleton rows falls into the rendering context.
+This loading strategy is often referred to as [**viewport loading**](#viewport-loading).
+
+If the total row count is unknown, the Data Grid fetches more data when the user scrolls to the bottom. This loading strategy is often referred to as [**infinite loading**](#infinite-loading).
+
+:::info
+Row count can be provided in either of the following ways.
+
+- Pass as [`rowCount`](/x/api/data-grid/data-grid/#data-grid-prop-rowCount) prop
+- Return `rowCount` in the `getRows` method of the [data source](/x/react-data-grid/server-side-data/#data-source)
+- Set the `rowCount` using the [`setRowCount`](/x/api/data-grid/grid-api/#grid-api-prop-setRowCount) API method.
+
+The above list is given in the order of precedence, which means if the row count is set using the API, that value gets overridden once a new value is returned by the `getRows` method, unless if it is `undefined`.
+:::
+
+## Viewport loading
+
+The viewport loading mode is enabled when the row count is known (`rowCount >= 0`). Grid fetches the first page immediately and adds skeleton rows to match the total row count. Other pages are fetched once the user starts scrolling and moves a skeleton row inside the rendering context (index range defined by [Virtualization](/x/react-data-grid/virtualization/)).
+
+If the user scrolls too fast, the grid loads multiple pages with one request (by adjusting `start` and `end` param) in order to reduce the server load.
+
+The demo below shows the viewport loading mode.
+
+{{"demo": "ServerSideLazyLoadingViewport.js", "bg": "inline"}}
+
+:::info
+The data source demos use a utility function `useMockServer` to simulate the server-side data fetching.
```

Comment:
```suggestion
The data source demos use a `useMockServer` utility function to simulate server-side data fetching.
```

Reply:
This is copied from the other page, so I will update other occurrences as well